docs/databases/users-roles-privileges.md

```diff
@@ -0,0 +1,192 @@
+
+The structure of your database privileges should fit into your [permissions strategy](https://www.metabase.com/learn/permissions/strategy).
+
+## Organize database privileges with roles
+
+Database roles can be used to bundle database privileges for different use cases. One user can belong to multiple roles and get the total set of privileges across those roles.
+
+We recommend creating a `metabase` user with the following roles:
+
+- [`analytics` for read access](#granting-read-access-to-analytics-tables) to any schemas or tables used for analysis.
+- Optional [`metabase_actions` for write access](#actions) to tables used for Metabase actions.
+- Optional [`metabase_model_caching` for write access](#model-caching) to the schema used for Metabase model caching.
+
+Bundling your privileges into roles based on use cases makes it easier to manage privileges in the future (especially in [multi-tenant situations](#multi-tenant-permissions)). For example, you could:
+
+- Use the same `analytics` role for other BI tools in your [data stack](https://www.metabase.com/learn/databases/data-landscape#data-analysis-layer) that need read-only access to the analytics tables in your database.
+- Revoke the write access for `metabase_model_caching` without affecting the write access for `metabase_actions`.
+
+## Granting read access to analytics tables
+
+Most analytics tools will need the same [minimum database privileges](#minimum-database-privileges) to connect and read from analytics tables in your database.
+
+To organize those minimum privileges, you'll:
+
+- Create a database role called `analytics`.
+- Create a database user called `metabase`.
+- Add `metabase` to the `analytics` role.
+- Add privileges to the `analytics` role.
+
+For example, if you're using a Postgres database, you'd log in as an admin and run the SQL statements:
+
+```sql
+-- Create a role named "analytics".
+CREATE ROLE analytics WITH LOGIN;
+
+-- Add the CONNECT privilege to the role.
+GRANT CONNECT ON DATABASE "your_database" TO analytics;
+
+-- Create a database user named "metabase".
+CREATE USER metabase WITH PASSWORD "your_password";
+
+-- Give the role to the metabase user.
+GRANT analytics TO metabase;
+
+-- Add query privileges to the role (options 1-3):
+-- Option 1: Let users with the analytics role query anything in the DATABASE.
+GRANT SELECT ON DATABASE "your_database" TO analytics;
+
+-- Option 2: Let users with the analytics role query anything in a specific SCHEMA.
+-- GRANT SELECT ON ALL TABLES IN SCHEMA "schema" TO analytics;
+
+-- Option 3: Let users with the analytics role query anything in a specific TABLE.
+-- GRANT SELECT ON ALL TABLES IN SCHEMA "schema" TO analytics;
+```
+
+Remember that when you grant privileges to a role, all users with that role will get those privileges.
+
+## Minimum database privileges
+
+In order to view and query your tables in Metabase, you'll have to give Metabase's database user:
```

Comment:
Can we get the SQL here for the lazy (following the grant to role, then grant role to user pattern)?